compiler/rustc_ty_utils/src/abi.rs

```diff
@@ -786,7 +786,7 @@ fn fn_abi_adjust_for_abi<'tcx>(
                     // an LLVM aggregate type for this leads to bad optimizations,
                     // so we pick an appropriately sized integer type instead.
                     arg.cast_to(Reg { kind: RegKind::Integer, size });
-                } else if size <= data_pointer_size * 2 && size.bytes() % 2 == 0 {
+                } else if size == data_pointer_size * 2 && size.bytes() % 2 == 0 {
```

Comment:
There's no comment or anything explaining this subtlety -- please add it.

Reply:
Yeah, my original idea on how this should look got shot down by CI, so I'll look into this some more and add comments to the final version.

Reply:
Right, so it would be useful to understand *why* the original assumption you made is not compatible with codegen and explain it here. It also may suggest a deeper bug in your implementation, though it's not certain unless you look into it.